src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -401,6 +406,33 @@ public CatalogObjectMetadata restore(String bucketName, String name, Long commit
         return new CatalogObjectMetadata(restoredRevision);
     }
 
+    /**
+     * Method returns all stored kinds for all objects in catalog
+     * @return unique set of kinds with root kind
+     * for example kinds: a/b, a/c, d/f/g
+     * should return a/b, a/c, d/f/g, a
+     */
+    public Set<String> getKinds() {
```

Comment:
It is not very readable  and time complexity seems slow (O(AxB) A is the size of allStoredKinds and B splittedKinds size). 
This line:
Set<String> resultKinds = new HashSet<>(allStoredKinds);

Is populating the resultKinds with all values of allStoredKinds right? why? 

Lets work together on refactoring this part.